File: csharp/prime-factors/PrimeFactors.cs
using System;
using System.Collections.Generic;

public static class PrimeFactors
{
    public static int[] Factors(long number)
    {
        var list  = new List<int>();
        int division = 2;

        do
        {
            if(number % division == 0)
            {
                list.Add(division);
                number = number / division;
                division = 2; 
            } 
            else division++;                
        } 
        while (number > 1);

        return list.ToArray();
    }
}

File: javascript/prime-factors/prime-factors.js
//
// This is only a SKELETON file for the 'Prime Factors' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const primeFactors = (number) => {
  let list = [];
  let division = 2;

  do
  {
      if(number % division == 0)
      {
          list.push(division);
          number = number / division;
          division = 2; 
      } 
      else division++;                
  } 
  while (number > 1);

  return list;
};
